kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/http/AbstractSimultaneousConnectionsTest.java

```diff
@@ -156,41 +151,90 @@ public void http1WebSocketConnections() throws Exception {
     withHttp1();
     final Collection<okhttp3.WebSocket> serverSockets = ConcurrentHashMap.newKeySet();
     final Collection<WebSocket> clientSockets = ConcurrentHashMap.newKeySet();
-    final CountDownLatch latch = new CountDownLatch(MAX_HTTP_1_CONNECTIONS);
+    final CyclicBarrier cyclicBarrier = new CyclicBarrier(2);
+    final CountDownLatch latch = new CountDownLatch(MAX_HTTP_1_WS_CONNECTIONS);
     final MockResponse response = new MockResponse()
         .withWebSocketUpgrade(new WebSocketListener() {
           @Override
           public void onOpen(okhttp3.WebSocket webSocket, Response response) {
+            try {
+              cyclicBarrier.await(1, TimeUnit.SECONDS);
+            } catch (Exception ignore) {
+            }
             serverSockets.add(webSocket);
             webSocket.send("go on");
           }
         });
-    IntStream.range(0, MAX_HTTP_1_CONNECTIONS).forEach(i -> mockWebServer.enqueue(response));
+    IntStream.range(0, MAX_HTTP_1_WS_CONNECTIONS).forEach(i -> mockWebServer.enqueue(response));
     try (final HttpClient client = clientBuilder.build()) {
-      for (int it = 0; it < MAX_HTTP_1_CONNECTIONS; it++) {
+      for (int it = 0; it < MAX_HTTP_1_WS_CONNECTIONS; it++) {
         client.newWebSocketBuilder()
             .uri(mockWebServer.url("/").uri())
             .buildAsync(new WebSocket.Listener() {
 
               @Override
               public void onMessage(WebSocket webSocket, String text) {
+                clientSockets.add(webSocket);
                 latch.countDown();
                 webSocket.request();
               }
-            })
-            .whenComplete((ws, t) -> clientSockets.add(ws));
+            });
+        cyclicBarrier.await(1, TimeUnit.SECONDS);
       }
       assertThat(latch.await(20L, TimeUnit.SECONDS)).isTrue();
       assertThat(serverSockets.size())
-          .isEqualTo(MAX_HTTP_1_CONNECTIONS)
+          .isEqualTo(MAX_HTTP_1_WS_CONNECTIONS)
           .isLessThanOrEqualTo((int) serverSocketFactory.activeConnections());
+      assertThat(clientSockets)
+          .hasSize(MAX_HTTP_1_WS_CONNECTIONS);
     } finally {
       for (okhttp3.WebSocket socket : serverSockets) {
         socket.close(1000, "done");
       }
     }
   }
 
+  private static class DelayedResponseHandler implements HttpHandler {
+
+    private final int requestCount;
+    private final CyclicBarrier barrier;
+    private final Set<HttpExchange> exchanges;
+    private final AtomicInteger connectionCount;
+    private final CompletableFuture<Void> done;
+
+    private DelayedResponseHandler(int requestCount, CyclicBarrier barrier, HttpHandler handler) {
+      this.requestCount = requestCount;
+      this.barrier = barrier;
+      exchanges = ConcurrentHashMap.newKeySet();
+      connectionCount = new AtomicInteger(0);
+      done = new CompletableFuture<>();
+      done.thenRunAsync(() -> {
```

Comment:
In local testing a saw two types of errors on the client side - 500s from before the request is ever processed by the server, and errors after createContext has been called, but before the headers are sent.  That is it appeared that we could flood the output buffering of the HttpServer.  I also didn't see at first glance if this was a non-blocking operation, so for these reasons used a bounded theadpool, rather than relying for this rather than relying on the common pool.